### PR TITLE
Make callouts not close the reveal they are in.

### PR DESCRIPTION
**Pull request by Owlbertz**
This solution is kinda sloppy, maybe there is a more elegant way to solve this...
Hope someone else can review this soon, really would like to see this in the upcoming release.

### CODE REVIEW

**Comment by Silvester20** [comment]
This will fix https://github.com/zurb/foundation-sites/issues/8428, so many thanks for fixing this!

**Comment by kball** [comment]
Hmm... this approach feels like it could have unintended side effects.  What if instead we have the callout event handler call event.stopPropagation() to prevent it bubbling up?

**Comment by Owlbertz** [comment]
Well there is no callout event handler because it's all done in the `Foundation.util.triggers.js`:
```js
(document).on('close.zf.trigger', '[data-closable]', function(e){
  e.stopPropagation();
  let animation = $(this).data('closable');

  if(animation !== ''){
    Foundation.Motion.animateOut($(this), animation, function() {
      $(this).trigger('closed.zf');
    });
  }else{
    $(this).fadeOut().trigger('closed.zf');
  }
});
```
There is already a `stopPropagation()`, but that doesn't seem to work well here. That's why I built it this way.

**Comment by kball** [comment]
Ah interesting... the reveal handler is probably getting called first, given that the callout doesn't have its own handler but is depending on that base handler.

Hmm... so I still don't think this is the right fix, because it will break the ability to simply include a 'data-close' element inside a reveal... but I'm not sure what the right fix is.  I'm also not 100% sure that element is going to get passed along in all click cases...

What we're really trying to do is handle nesting... so what if instead of looking at element we did a check something like this

```
if ($(event.target).parents('[data-closable]')[0] !== this.$element[0])
```

Basically looking to see if there's another data-closable element in the tree before you get to the reveal event handler?

**Comment by Silvester20** [comment]
Any updates on this?

**Comment by kball** [comment]
@Owlbertz can you take a look at the concern above and rework this fix to still allow using data-close inside of a reveal?

**Comment by kball** [comment]
Great.  Nice work!

**Comment by rui-castro** [comment]
This bug still exists in foundation 6.4.3.

Using something like the example bellow; clicking callout close button closes the reveal.

```html
<div class="reveal reveal-animated">

        <div class="callout warning" data-closable>
            <button class="close-button" aria-label="Close warning" type="button" data-close>
                <span aria-hidden="true">&times;</span>
            </button>
            Callout text
        </div>

    <a aria-label="Close" data-close>&#215;</a>
</div>
```
